src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -89,8 +91,18 @@ func (r *ClusterReconciler) Reconcile(
 
 	var observedPods corev1.PodList
 
-	err := r.List(ctx, &observedPods, &client.ListOptions{
-		LabelSelector:	labels.ForCluster(&redpandaCluster).AsClientSelector(),
+	newReq, err := labels2.NewRequirement(
+		appsv1.ControllerRevisionHashLabelKey,
+		selection.Equals,
+		[]string{sts.LastObservedState.Status.CurrentRevision},
```

Comment:
This strikes me as trying to fix a problem at the wrong layer. Admission control (mentioned above) seems related to what this is attempting to address, or simply accepting that users _can_ hurt themselves by not following instructions. But I would tend to agree with @alenkacz that we should be relying on standard label selectors at this level.

The obvious caveat here being that I am a k8s layperson.